arch/xtensa/src/esp32/esp32_ble_adapter.c

```diff
@@ -0,0 +1,2814 @@
+                         adapter_coex_schm_register_btdm_callback,
+  ._coex_bb_reset_lock = coex_bb_reset_lock_wrapper,
+  ._coex_bb_reset_unlock = coex_bb_reset_unlock_wrapper,
+};
+
+/* The mode column will be modified by release function to indicate the
+ * available region
+ */
+
+static btdm_dram_available_region_t btdm_dram_available_region[] =
+{
+  /* following is .data */
+
+  {
+    ESP_BT_MODE_BTDM,
+    SOC_MEM_BT_DATA_START,
+    SOC_MEM_BT_DATA_END
+  },
+
+  /* following is memory which HW will use */
+
+  {
+    ESP_BT_MODE_BTDM,
+    SOC_MEM_BT_EM_BTDM0_START,
+    SOC_MEM_BT_EM_BTDM0_END
+  },
+  {
+    ESP_BT_MODE_BLE,
+    SOC_MEM_BT_EM_BLE_START,
+    SOC_MEM_BT_EM_BLE_END
+  },
+  {
+    ESP_BT_MODE_BTDM,
+    SOC_MEM_BT_EM_BTDM1_START,
+    SOC_MEM_BT_EM_BTDM1_END
+  },
+  {
+    ESP_BT_MODE_CLASSIC_BT,
+    SOC_MEM_BT_EM_BREDR_START,
+    SOC_MEM_BT_EM_BREDR_REAL_END
+  },
+
+  /* following is .bss */
+
+  {
+    ESP_BT_MODE_BTDM,
+    SOC_MEM_BT_BSS_START,
+    SOC_MEM_BT_BSS_END
+  },
+  {
+    ESP_BT_MODE_BTDM,
+    SOC_MEM_BT_MISC_START,
+    SOC_MEM_BT_MISC_END
+  },
+};
+
+/****************************************************************************
+ * Private Functions and Public Functions only used by libraries
+ ****************************************************************************/
+
+static int adapter_coex_register_bt_cb_wrapper(coex_func_cb_t cb)
+{
+  wlinfo("I'm here!\n");
+  return ESP_ERR_INVALID_STATE;
+}
+
+static int adapter_coex_schm_register_btdm_callback(void *callback)
+{
+  wlinfo("I'm here!\n");
+  return ESP_ERR_INVALID_STATE;
+}
+
+static int adapter_coex_register_wifi_channel_change_callback(void *cb)
+{
+  wlinfo("I'm here!\n");
+  return ESP_ERR_INVALID_STATE;
+}
+
+static int adapter_coex_wifi_channel_get(uint8_t *primary,
+                                         uint8_t *secondary)
+{
+  wlinfo("I'm here!\n");
+  return -1;
+}
+
+static void adapter_coex_schm_status_bit_clear(uint32_t type,
+                                               uint32_t status)
+{
+  wlinfo("I'm here!\n");
+}
+
+static void adapter_coex_schm_status_bit_set(uint32_t type, uint32_t status)
+{
+  wlinfo("I'm here!\n");
+}
+
+static uint32_t adapter_coex_schm_interval_get(void)
+{
+  wlinfo("I'm here!\n");
+  return ESP_ERR_INVALID_STATE;
+}
+
+static uint8_t adapter_coex_schm_curr_period_get(void)
+{
+  wlinfo("I'm here!\n");
+  return 0;
+}
+
+static void *adapter_coex_schm_curr_phase_get(void)
+{
+  wlinfo("I'm here!\n");
+  return NULL;
+}
+
+/****************************************************************************
+ * Name: esp_errno_trans
+ *
+ * Description:
+ *   Transform from nuttx error code to Wi-Fi adapter error code
+ *
+ * Input Parameters:
+ *   ret - NuttX error code
+ *
+ * Returned Value:
+ *   Wi-Fi adapter error code
+ *
+ ****************************************************************************/
+
+static inline int32_t esp_errno_trans(int ret)
```

Comment:
```suggestion
static inline bool esp_errno_trans(int ret)
```
Are Wi-Fi error codes simply false and true?

Reply:
I cannot convert it to bool because it is returned by many wrapper function that aren't expecting boolean type!